File: apps/executable-spec-web/sdk/js/examples/quickstart.js
import { NormPegClient } from "../normpeg-sdk.js";

async function main() {
  const client = new NormPegClient({
    baseUrl: "http://localhost:8790",
    role: "admin",
    actorId: "demo-js-sdk",
    tenantId: "default",
  });

  // 1) Register spec from markdown.
  const registerResult = await client.registerSpec({
    markdown: `
# Demo Spec
input: value
rule: result >= 90
`,
  });
  console.log("register result:", registerResult);

  // 2) Execute.
  const executeResult = await client.execute({
    spuId: "demo.spu@v1",
    containerId: "container_001",
    inputs: { value: 95 },
  });
  console.log("execute result:", executeResult);

  // 3) Query proof (container must be archived in workflow).
  const proofResult = await client.queryProof("container_001");
  console.log("proof result:", proofResult);
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: apps/executable-spec-web/sdk/python/examples/quickstart.py
from pathlib import Path
import sys

sys.path.append(str(Path(__file__).resolve().parents[1]))

from normpeg_sdk import NormPegClient


def main() -> None:
    client = NormPegClient(
        base_url="http://localhost:8790",
        role="admin",
        actor_id="demo-python-sdk",
        tenant_id="default",
    )

    # 1) Register spec from markdown.
    register_result = client.register_spec(
        markdown="""
# Demo Spec
input: value
rule: result >= 90
"""
    )
    print("register result:", register_result)

    # 2) Execute.
    execute_result = client.execute(
        {
            "spuId": "demo.spu@v1",
            "containerId": "container_001",
            "inputs": {"value": 95},
        }
    )
    print("execute result:", execute_result)

    # 3) Query proof (container must be archived in workflow).
    proof_result = client.query_proof("container_001")
    print("proof result:", proof_result)


if __name__ == "__main__":
    main()
